nautobot/core/views/generic.py

```diff
@@ -47,6 +47,34 @@
 from nautobot.utilities.views import GetReturnURLMixin, ObjectPermissionRequiredMixin
 
 
+def get_relationships_errors(request, obj):
+    required_relationships = Relationship.objects.get_required_for_model(obj)
+    relationships_errors = []
+    for side, relations in required_relationships.items():
+        for relation in relations:
+
+            # Skip referencing itself
+            if ContentType.objects.get_for_model(obj) == getattr(relation, f"{relation.required}_type"):
+                continue
+
+            model_class = getattr(relation, f"{side}_type").model_class()
+
+            if not model_class.objects.count():
+                model_meta = model_class._meta
+                required = model_meta.verbose_name
+                try:
+                    add_url = reverse(f"{model_meta.app_label}:{model_meta.model_name}_add")
```

Comment:
Should be able to use `nautobot.utilities.utils.get_route_for_model` here (possibly dependent on #2223?)

Reply:
Do we know when #2223 is likely to be merged in?